src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -69,6 +79,96 @@ private String setKey() {
+   */
+
+  private String formatDate(final String date) {
+    return date.substring(0, 4) + "-" + date.substring(4, 6) + "-" + date.substring(6);
+  }
+
+  /**
+   * Gets historical price data for the given symbol and date from the IEX API. Saves the new
+   * IexHistoricalPrice object to the price repository as a side effect.
+   */
+
+  private List<IexHistoricalPrice> getHistoricalPriceForDate(
+      final String symbol,
+      final String date) {
+
+    String searchDate = formatDate(date);
+    log.info("Searching repository for symbol {} and date {}", symbol, searchDate);
+
+    if (priceRpsy.existsBySymbolAndDate(symbol, searchDate)) {
+      log.info("Historical price for symbol {}, date{} found in repository.", symbol, date);
+      return priceRpsy.findBySymbolAndDate(symbol, searchDate);
+    } else {
+      log.info("Requesting historical price for symbol {}, date {} from IEX.", symbol, date);
+      List<IexHistoricalPrice> prices;
+      prices = iexClient.getHistoricalPriceForDate(symbol, date, true, key);
+      Iterable<IexHistoricalPrice> saved = priceRpsy.saveAll(prices);
+      log.info("Saved historical price to repository: {}", saved);
+      return prices;
+    }
+  }
+
+  private LocalDate rangeToStartDateHelper(final String range, final LocalDate today) {
+
+    ChronoUnit unit = ChronoUnit.DAYS;
+    String rangeSuffix = range.substring(range.length() - 1);
+
+    if (rangeSuffix.equals("m")) {
+      unit = ChronoUnit.MONTHS;
+    } else if (rangeSuffix.equals("y")) {
+      unit = ChronoUnit.YEARS;
+    } else if (!rangeSuffix.equals("d")) {
+      log.warn("Argument range must end with \"d\", \"m\", or \"y\"; received {}.", rangeSuffix);
+      log.warn("Assuming units of DAYS.");
+    }
+
+    int amountToSubtract = Integer.parseInt(range.substring(0, range.length() - 1));
+    return today.minus(amountToSubtract, unit);
+  }
+
+  /**
+   * Takes an IEX range and returns the first date in the range.
+   * @param range A string representing an IEX range for some number of days, months, or years.
+   * @return A LocalDate object representing the first date in the range.
+   */
+  public LocalDate rangeToStartDate(final String range) {
+
+    return rangeToStartDateHelper(range, LocalDate.now());
+  }
+
+  /**
+   * Gets historical price data for the given symbol and range from the IEX API. Saves new
+   * IexHistoricalPrice objects to the price repository as a side effect.
+   */
+
+  private List<IexHistoricalPrice> getHistoricalPricesForRange(
+      final String symbol,
+      final String range) {
+
+    LocalDate rangeStartDate = rangeToStartDate(range);
+    if (earliestRequestedRangeStart != null
+        && !rangeStartDate.isBefore(earliestRequestedRangeStart)) {
+      log.info("Historical prices for range {} found in repository.", range);
+      return priceRpsy.findBySymbolAndDateGreaterThanEqual(symbol,
+          rangeStartDate.toString());
```

Comment:
Try to avoid using toString(). Can be dangerous, but also unnecessary in most cases